core/job/service/job_service.go

```diff
@@ -561,21 +630,35 @@ func (j *JobService) Validate(ctx context.Context, jobTenant tenant.Tenant, jobS
 	}
 
 	err = job.Specs(jobSpecs).Validate()
-	me.Append(err)
-	validatedJobSpecs := job.Specs(jobSpecs).GetValid()
+	if err != nil {
+		return err
+	}
 
+	validatedJobSpecs := job.Specs(jobSpecs).GetValid()
 	existingJobs, err := j.jobRepo.GetAllByTenant(ctx, jobTenant)
-	me.Append(err)
-
-	toAdd, toUpdate, toDelete, unmodifiedSpecs, err := j.differentiateSpecs(existingJobs, validatedJobSpecs, jobNamesWithInvalidSpec)
-	logWriter.Write(writer.LogLevelInfo, fmt.Sprintf("[%s] found %d new, %d modified, and %d deleted job specs", jobTenant.NamespaceName().String(), len(toAdd), len(toUpdate), len(toDelete)))
-	me.Append(err)
+	if err != nil {
+		return err
+	}
 
-	if len(toAdd)+len(toUpdate)+len(toDelete) == 0 {
+	toAdd, toUpdate, toDelete, unmodifiedSpecs, unmodifiedDirtySpecs := j.differentiateSpecs(existingJobs, validatedJobSpecs, jobNamesWithInvalidSpec)
+	logWriter.Write(writer.LogLevelInfo, fmt.Sprintf("[%s] found %d new, %d modified, %d deleted and %d dirty job specs", jobTenant.NamespaceName().String(), len(toAdd), len(toUpdate), len(toDelete), len(unmodifiedDirtySpecs)))
+	if len(unmodifiedDirtySpecs) > 0 {
+		var dirtyJobsName []string
+		for _, job := range unmodifiedDirtySpecs {
+			dirtyJobsName = append(dirtyJobsName, job.Name().String())
+		}
+		errorMsg := fmt.Sprintf("Found %d dirty jobs for tenant: %s/%s, Jobs: %s", len(unmodifiedDirtySpecs), jobTenant.ProjectName(), jobTenant.NamespaceName(), dirtyJobsName)
```

Comment:
rename Dirty as `Unprocessed`

Reply:
there are unprocessed jobs in this namespace, they will be reprocessed as part of replace_all of this Pipeline 
( same error msg in Replace all )